### ISSUE TITLE
HA 2024.1 - Multiple deprecated constants

### ISSUE THREAD

**Issue by Troon** [msp1974/homeassistant-jlrincontrol#106]
From the logs on my 2024.1 install:
```
SOURCE_TYPE_GPS was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use SourceType.GPS instead, please report it to the author of the 'jlrincontrol' custom integration
PRESSURE_BAR was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPressure.BAR instead, please report it to the author of the 'jlrincontrol' custom integration
```

**Comment by ismarslomic** [comment]
Thanks for reporting this, @Troon! I will look at this in upcoming days and release new version of the integration

**Comment by ismarslomic** [comment]
Additional constants that are deprecated (seen in `HA 2024.1.2`)

1. `PRESSURE_BAR` was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPressure.BAR instead, please report it to the author of the 'jlrincontrol' custom integration

2. `PRESSURE_PSI` was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPressure.PSI instead, please report it to the author of the 'jlrincontrol' custom integration

3. `LENGTH_KILOMETERS` was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfLength.KILOMETERS instead, please report it to the author of the 'jlrincontrol' custom integration

4. `LENGTH_METERS` was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfLength.METERS instead, please report it to the author of the 'jlrincontrol' custom integration

5. `PRESSURE_PA` was used from jlrincontrol, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPressure.PA instead, please report it to the author of the 'jlrincontrol' custom integration

**Comment by ismarslomic** [comment]
@msp1974 just a clarification regarding the [homeassistant](https://github.com/msp1974/homeassistant-jlrincontrol/blob/master/hacs.json#L5) key in the hacs manifest file [hacs.json](https://github.com/msp1974/homeassistant-jlrincontrol/blob/master/hacs.json). 

When we change to these new constants we would also break the backward compatibility, and would need to adjust the homeassistant version that is required for new version of the JLR integration. Right?

**Comment by msp1974** [comment]
Yes that's correct.  Not sure when these constants were added but we should make it earliest version that they are in.

**Comment by ismarslomic** [comment]
After checking quickly in HA core, it seems that some of these constants has existing for some time, but the deprecation warnings were introduced in `v2024.1`.

Btw: I see that the manifest file has not been touched for 4 years, but we do specify that minimum required HA version is 2022.10 in README. So I guess it just a miss from our side?

> Due to changes in Home Assistant, this integration requires a minimum of HA v2022.10

**Comment by Troon** [comment]
`UnitOfPressure` enum introduced in https://github.com/home-assistant/core/pull/81009
`UnitOfLength` enum introduced in https://github.com/home-assistant/core/pull/81011

These constants were introduced in `2022.11`.

**Comment by ismarslomic** [comment]
`SourceType` enum introduced in https://github.com/home-assistant/core/pull/75892 and HA `2022.9`
`UnitOfTemperature`  enum introduced in https://github.com/home-assistant/core/pull/81006 and HA `2022.11`
`UnitOfVolume` enum introduced in https://github.com/home-assistant/core/pull/81028 and HA `2022.11`
`UnitOfEnergy` enum introduced in https://github.com/home-assistant/core/pull/80998 and HA `2022.11`

**Comment by ismarslomic** [comment]
Resolved in `v2.2.5`
